ledger/cow.go

```diff
@@ -68,15 +69,19 @@ type roundCowState struct {
 	compatibilityMode bool
 	// cache mainaining accountIdx used in getKey for local keys access
 	compatibilityGetKeyCache map[basics.Address]map[storagePtr]uint64
+
+	// track creatables created during each transaction in the round
+	trackedCreatables map[int]basics.CreatableLocator
 }
 
 func makeRoundCowState(b roundCowParent, hdr bookkeeping.BlockHeader, prevTimestamp int64, hint int) *roundCowState {
 	cb := roundCowState{
-		lookupParent: b,
-		commitParent: nil,
-		proto:        config.Consensus[hdr.CurrentProtocol],
-		mods:         ledgercore.MakeStateDelta(&hdr, prevTimestamp, hint, 0),
-		sdeltas:      make(map[basics.Address]map[storagePtr]*storageDelta),
+		lookupParent:      b,
+		commitParent:      nil,
+		proto:             config.Consensus[hdr.CurrentProtocol],
+		mods:              ledgercore.MakeStateDelta(&hdr, prevTimestamp, hint, 0),
+		sdeltas:           make(map[basics.Address]map[storagePtr]*storageDelta),
+		trackedCreatables: make(map[int]basics.CreatableLocator),
```

Comment:
why `CreatableLocator` and not `CreatableID`?

Reply:
Yeah, I think the CreatableLocator is the wrong thing.  (I think it goes back to a model in which we thought asas and app would be named by a creator and an account local index.). I don't think the Index in this thing is the appId, it's 5 if it's the user's 6th app created. I see your tests check that it's > 0, but you need to somehow test that it's a valid appid.

You might be able to compare it to `global CurrentApplicationID` in the creating teal code. I think that value is available, even during creation.  And depending on where that comes from, maybe that's an easier way to implement this.

Reply:
yes, `CreatableIndex`

Reply:
Figured I would try to make it future-proof with a low overhead cost in case we wanted to expose other creatable properties to TEAL later, but I'll change it.